code/modules/admin/topic/topic_teleports.dm

```diff
@@ -136,7 +136,7 @@
 					to_chat(owner, SPAN_ALERT("Hive choice error. Aborting."))
 					return
 				var/datum/hive_status/Hive = GLOB.hive_datum[hives[faction]]
-				var/list/targets = Hive.totalXenos
+				var/list/targets = Hive.total_living_xenos
```

Comment:
completely unrelated to your refactor really but this needs a .Copy() at the end, as otherwise this ends up mutating the actual list